view/components/utils/totals-in-matrix.js

```diff
@@ -83,11 +88,12 @@ module.exports = function (config) {
 			$.forEach(verticalTotals, function (total) {
 				var subTotatals = [];
 				$.forEach(total.paths, function (path) {
+					if (!path) return;
 					subTotatals.push($('matrix-display-input-' + path));
 				});
 				fullTotal.subTotals = fullTotal.subTotals.concat(subTotatals);
 			});
-			totals.push(fullTotal);
+			if (!skipVerticalTotal) totals.push(fullTotal);
```

Comment:
So, if I see correctly, whenever I want to skipVerticalTotal, I will also skip total. This is weird, I thought you only wanted to skip vertical total not the whole total.

I believe this line is not needed because:

1. If skipVerticalTotal === true, then my verticalTotals is equal [] (so I can safaly update totals with fullTotal, as fullTotal may only have anything interesting in horizontalTotals).
2. If skipVerticalTotal is falsy, then it will work exactly as before this change.

Since you make sure that `skipVerticalTotal` will result in empty array passed as `verticalTotals`, then I don't think you even have to pass `skipVerticalTotal` to `$.onEnvUpdate`.